core/src/style/pointStyleBuilder.cpp

```diff
@@ -329,39 +299,106 @@ bool PointStyleBuilder::getUVQuad(Parameters& _params, glm::vec4& _quad, Texture
 
     if (texture) {
         *_texture = texture;
+    }
 
-        if (auto& atlas = texture->spriteAtlas()) {
+    return true;
+}
 
-            SpriteNode spriteNode;
+bool PointStyleBuilder::evalSizeParam(const DrawRule& _rule, Parameters& _params, const Texture* _texture) const {
+    StyleParam::SizeValue size;
+    SpriteNode spriteNode;
+    glm::vec2 cssSize = {1.f, 1.f};
+    float aspectRatio = 1.f;
+    bool useTextureInfo = false;
+
+    if (_texture) {
+        cssSize = glm::vec2{_texture->getWidth(), _texture->getHeight()} * _texture->invDensity();
+        aspectRatio = _texture->getWidth() / _texture->getHeight();
+        // can use cssSize and aspect if "%" or "auto" size is used
+        useTextureInfo = true;
+
+        const auto &atlas = _texture->spriteAtlas();
+        if (atlas) {
             if (!atlas->getSpriteNode(_params.sprite, spriteNode) &&
                 !atlas->getSpriteNode(_params.spriteDefault, spriteNode)) {
                 return false;
             }
+            cssSize = spriteNode.m_size * _texture->invDensity();
+            aspectRatio = spriteNode.m_size.x / spriteNode.m_size.y;
+        } else {
+            // missing sprite atlas for texture but sprite specified in draw rule
```

Comment:
I'm not clear how this determines that a sprite was specified in the draw rule.

Reply:
Yup this is incorrect, will update.

Reply:
Yeah I missed to update this earlier. Fixed now.